plugins/woocommerce/templates/content-product.php

```diff
@@ -20,7 +20,7 @@
 global $product;
 
 // Ensure visibility.
-if ( empty( $product ) || ! $product->is_visible() ) {
+if ( empty( $product ) || ! is_a( $product, 'WC_Product' ) || ! $product->is_visible() ) {
```

Comment:
Not sure we need to check it is empty **and** that it is a `WC_Product`. I'm also wondering if it might be better to reference the class in a slightly different way, to improve the debug+development experience for folks working with IDEs 🤔 

What do you think of:

```suggestion
if ( ! is_a( $product, WC_Product::class ) || ! $product->is_visible() ) {
```

Reply:
@barryhughes Thanks for reviewing this PR and sharing the valuable feedback. I have implemented the suggestions so could you please take a look into this.

Reply:
Hi @narenin, just a note, a `empty()` or `isset()` check is not needed here as the global statement initializes the var if not present in the $GLOBALS array. Not sure if defensively we should be using the check anyway or the null coalescing operator in case the way $product var arrives there to prevent a warning/notice oh PHP 8+/7.4

I've pre-approved the PR, but let's also get @barryhughes' take to ensure this is just being overly cautious ;)

Reply:
The suggestion was adopted in https://github.com/woocommerce/woocommerce/pull/51286/commits/9f706ae8b50a3cb4213af6585417c220345363c9, so I think we're good (we now know that `$product` is an instance of `WC_Product`) and things can flow from there. Per previous notes, though, we do still need a bump to the `@version` tag, testing instructions and a changelog.

Reply:
Thanks @wavvves @barryhughes for your feedback, I have updated the `@version` tag and also testing instructions and a changelog has been added.